### PR TITLE
Refactor distsql about shadow related syntax #21696

### LINKED ISSUE #21696: [DistSQL] Optimize `SHADOW` related syntax
Hi community,

This issue is for #21659.

### Aim
Optimize `SHADOW` related syntax, e.g.
```
Remove `CREATE SHADOW ALGORITHM`
Remove `ALTER SHADOW ALGORITHM`
Add `ALTER DEFAULT SHADOW ALGORITHM`, refer to `CREATE DEFAULT SHADOW ALGORITHM`
Add `SHOW DEFAULT SHADOW ALGORITHM`
```

### PR DESCRIPTION
**Pull request by jiangML**
Fixes #21696.

Changes proposed in this pull request:
- Remove: CREATE SHADOW ALGORITHM
- Remove: ALTER SHADOW ALGORITHM
- Add: SHOW DEFAULT SHADOW ALGORITHM
- Add: ALTER DEFAULT SHADOW ALGORITHM
- Refactor: CREATE DEFAULT SHADOW ALGORITHM NAME EQ algorithmName -> CREATE DEFAULT SHADOW ALGORITHM shadowAlgorithmDefinition

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `mvn clean install -B -T2C -DskipTests -Dmaven.javadoc.skip=true -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: features/shadow/distsql/statement/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/statement/ShowDefaultShadowAlgorithmStatement.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Show default shadow algorithm statement.
+ */
+@Getter
+public final class ShowDefaultShadowAlgorithmStatement extends ShowRulesStatement {
+    
+    public ShowDefaultShadowAlgorithmStatement(final DatabaseSegment database) {
+        super(database);
+    }
+
```
Comment (RaigorJiang) [inline features/shadow/distsql/statement/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/statement/ShowDefaultShadowAlgorithmStatement.java:33]: Please remove the useless blank line.

File: features/shadow/distsql/parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java [hunk 2]
```diff
@@ -173,4 +171,5 @@ public ASTNode visitDatabaseName(final DatabaseNameContext ctx) {
     public ASTNode visitCountShadowRule(final CountShadowRuleContext ctx) {
         return new CountShadowRuleStatement(Objects.nonNull(ctx.databaseName()) ? (DatabaseSegment) visit(ctx.databaseName()) : null);
     }
+
```
Comment (RaigorJiang) [inline features/shadow/distsql/parser/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/core/ShadowDistSQLStatementVisitor.java:174]: Please remove the useless blank line.

File: test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/distsql/rql/ShowDefaultShadowAlgorithmsStatementTestCase.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+package org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.statement.distsql.rql;
+
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.statement.DatabaseContainedTestCase;
+
+/**
+ * Show default shadow algorithm statement test case.
+ */
+public class ShowDefaultShadowAlgorithmsStatementTestCase extends DatabaseContainedTestCase {
+
```
Comment (RaigorJiang) [inline test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/distsql/rql/ShowDefaultShadowAlgorithmsStatementTestCase.java:26]: Please remove the useless blank line.

Comment (jiangML) [reply test/parser/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/distsql/rql/ShowDefaultShadowAlgorithmsStatementTestCase.java:26]: ok

File: features/shadow/distsql/statement/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/statement/ShowDefaultShadowAlgorithmStatement.java (hunk 1 above)
Comment (jiangML) [reply features/shadow/distsql/statement/src/main/java/org/apache/shardingsphere/shadow/distsql/parser/statement/ShowDefaultShadowAlgorithmStatement.java:33]: ok

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**